modules/controls.js

```diff
@@ -53,7 +53,7 @@ displaySystem.registerModule({
                     btn.innerHTML = fn;
                     btn.addEventListener('click',function() {
                         var data = inps.map(getValue);
-                        if (system.ws) {
+                        if (system.ws && system.connected) {
```

Comment:
Can't we just check on `system.connected` (i.e. leave out 'system.ws`)?
That way we become independent of the usage of ws here (e.g. if we start using mhub-client in the future).

Reply:
Not sure, as `sendMessage` actually depends on system.ws

Reply:
Of course, but that's an 'internal detail' of sendMessage. As in: this piece of code doesn't depend on it?

In fact, probably that should be refactored in the future to something like `system.sendMessage()`, not `system.ws.sendMessage()` (inside the `sendMessage()` in this file).

Reply:
Not really, since if there is no ws, we'd want to call the method directly.

What about abstracting all that stuff in a system function call `if (system.connected()) {}`? That way, all things can be checked in the system itself.

Reply:
If there is no ws, then connected will also be false, right?
That's why I think testing for `system.ws` is redundant.

If more complicated (i.e. dynamic) stuff is required in the future, it would even be possible to convert `system.connected` to a getter.